pkg/virt-controller/watch/vmi.go

```diff
@@ -424,7 +424,10 @@ func isPodReady(pod *k8sv1.Pod) bool {
 				return false
 			}
 		} else {
-			if containerStatus.Ready == false {
+			// When using istio the istio-proxy container will not be ready
+			// until there is a service pointing to this pod.
+			// We need to start the VM anyway
```

Comment:
Hi @davidvossel thanks for the comment 

> why do we need to start the VM anyway? is there a reason we shouldn't wait until there is a service point to the pod?

when we have a regular pod the application inside the pod will start even if all the containers are not ready.

If we don't add the check here the VM will not start until there is a service pointing to it but when we create a service it will take time until the application is ready (vm boot time and application initialize) and the is not the case for regular pods. So I wanted to preserve the same behavior like regular pods.

> In general, I don't feel comfortable with hardcoded exceptions like this. There's also no guarantee the sidecar will be called "istio-proxy"

I also don't like to do it but I don't want to implement a module of searching in the istio configmaps to find what is the name of the proxy. I prefer for now to document it (the use of hard coded proxy name)

Reply:
is there anything we can key off of besides name to detect the istio container more generically?

Reply:
I wasn't able to find anything else :(

Reply:
umm. okay. I don't want to block us on that then.

We'll need a unit test for this and make a note about this behavior in the user guide.

Reply:
done can you please review :)

Reply:
I will open a PR for the user guide